### PR TITLE
add tutorial: How to Create a Safe with the Same Address on a Different Network to Recover Funds

### PR DESCRIPTION
**Pull request by ademidun**
This guide will show you how to recreate a Safe with a specific address on a  different network. This is useful for recovering funds sent to a Safe that exists on one network but not another. This is is an issue that [many](https://ethereum.stackexchange.com/questions/141408/sent-usdc-to-wrong-chain-on-gnosis-need-to-recover) [users](https://ethereum.stackexchange.com/questions/129309/create-a-safe-with-the-same-address-on-the-bsc-chain) [have](https://ethereum.stackexchange.com/questions/127353/trying-to-recreate-a-safe-from-mainnet-into-gnosis-chain-issue) [faced](https://ethereum.stackexchange.com/search?q=%5Bgnosis-safe%5D+wrong+chain).

Note: The goal of deploying the same Safe address on different chains is to recover locked funds. We don't recommend using this method as a default way of having the same address on different chains. See [Account Abstraction in a Multichain Landscape - Part 1: Addresses](https://safe.mirror.xyz/4GcGAOFno-suTCjBewiYH4k4yXPDdIukC5woO5Bjc4w).

### CODE REVIEW

File: build/create-safe-same-address-different-network.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# How to Create a Safe with the Same Address on a Different Network to Recover Funds
+
+This guide will show you how to recreate a Safe with a specific address on a  different network. This is useful for recovering funds sent to a Safe that exists on one network but not another. This is is an issue that [many](https://ethereum.stackexchange.com/questions/141408/sent-usdc-to-wrong-chain-on-gnosis-need-to-recover) [users](https://ethereum.stackexchange.com/questions/129309/create-a-safe-with-the-same-address-on-the-bsc-chain) [have](https://ethereum.stackexchange.com/questions/127353/trying-to-recreate-a-safe-from-mainnet-into-gnosis-chain-issue) [faced](https://ethereum.stackexchange.com/search?q=%5Bgnosis-safe%5D+wrong+chain).
```
Comment (rmeissner) [inline build/create-safe-same-address-different-network.md:3]: The BSC case might not be covered by this guide.

File: build/create-safe-same-address-different-network.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+### How it Works
+
+Safes are created by calling [`createProxyWithNonce` which uses `CREATE2`](https://github.com/safe-global/safe-contracts/blob/ba526d0475e3004f9fcd71cd25ebbd225ebeee7f/contracts/proxies/SafeProxyFactory.sol#L32)
+
+The [CREATE2 opcode](https://eips.ethereum.org/EIPS/eip-1014) gives you the ability to predict the address where a contract will be deployed, without deploying it. See [Deploying Smart Contracts Using CREATE2](https://docs.openzeppelin.com/cli/2.8/deploying-with-create2) for more information about this.
+
+### Recreating a Safe from Ethereum Mainnet
+
+If you recreate a Safe that was originally created on Ethereum Mainnet not another chain, it won't show up in the UI.
```
Comment (rmeissner) [inline build/create-safe-same-address-different-network.md:77]: There are other cases where this is also the case. In general if the Safe does not use the singleton that we support on that specific chain. (i.e. on optimism there are multiple versions deployed, but we do not support all of them via our interface)

File: examples/utils/recreateSafe.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    // Replay the transaction that created the safe on the original chain on the recovery chain
+    const reCreateSafeTx = {
+        to: proxyFactoryAddress,
+        data: safeCreationTx.data,
+        gasPrice,
+        value: 0,
+    }
+
+    const owner1Signer = new ethers.Wallet(process.env.OWNER_1_PRIVATE_KEY!, recoveryChainProvider)
```
Comment (rmeissner) [inline examples/utils/recreateSafe.ts:64]: This doesn't need to be an owner key

File: build/create-safe-same-address-different-network.md (hunk 1 above)
Comment (Uxio0) [reply build/create-safe-same-address-different-network.md:3]: `a  different` : 2 spaces

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# safe-examples
```
[the remark this answers is not in the document]
Comment (ademidun) [reply examples/README.md:1]: This file is used to show how to run the scripts in `examples/` it actually addresses the other [comment](https://github.com/safe-global/safe-docs/pull/179#discussion_r1146075179) you made

> There are no instructions how to run this script

Note: This file is outdated and will most likely be [created](https://github.com/safe-global/safe-docs/pull/187/files#diff-49aaa2819e35a856818ecec8c9fa7e1c79ad028d3f44bd749736353cfb51bac9) in PR #118

**Pull request closed without merging**